lib/model/CDataCategorizer.cc

```diff
@@ -56,13 +58,23 @@ std::size_t CDataCategorizer::memoryUsage() const {
 }
 
 bool CDataCategorizer::addExample(int categoryId, const std::string& example) {
+    // Don't add examples if we're memory-limited
+    if (m_Limits.resourceMonitor().getMemoryStatus() != model_t::E_MemoryStatusOk) {
```

Comment:
Is there any reason not to use `this->areNewCategoriesAllowed()` here? I couldn't think of one.

Reply:
`areNewCategoriesAllowed()` returns `false` when the job is in `hard_limit`.  But we stop adding examples as soon as the job goes to `soft_limit`.  It's in the spec for memory limiting in #1130 and the PR description, but I will expand the comment in the code to say this.